pan/strings.pan

```diff
@@ -0,0 +1,22 @@
+declaration template pan/strings;
+
+
+@documentation{
+Convenience type specifying a string that must contain at least one character (anything other than an empty string).
+Intended for use in building more complex types.
+}
+type string_non_empty = string(1..);
+
+
+@documentation{
+Convenience type specifying a string that may not contain any whitespace.
+Intended for use in building more complex types.
+}
+type string_non_whitespace = string with !match(SELF, '\s');
```

Comment:
this allows empty string, not sure if you want that. if it is allowed, maybe clarify in documnetation

Reply:
Wasn't supposed to be allowed before your suggested change, will fix.